src/backend/joanie/core/models/products.py

```diff
@@ -307,16 +351,40 @@ def validate(self):
                     enrollment.is_active = True
                     enrollment.save()
 
+    def get_enrollments(self, is_active=None):
+        """
+        Retrieve owner's enrollments related to the order courses.
+        """
+        filters = {
+            "course_run__course__in": self.target_courses.all(),
```

Comment:
not sure this helps:
```suggestion
            "course_run__course__in": self.target_courses.only("pk").all(),
```